enumerable.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def my_each
+    if self.class == Range
+      l = to_a
+      count = 0
+      while count < l.length
+        yield(l[count])
+        count += 1
+      end
+      l
+    elsif self.class == Hash
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    else
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
+    array = []
+    my_each do |i|
+      array.push(i) if yield(i)
+    end
+    array
+  end
+
+  def my_all?
+    my_each do |i|
+      (return true if yield(i) == true)
+      (return false if yield(i) == false)
+    end
+    true
+  end
+
+  def my_any?
```

Comment:
* [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

```  p %w[ant bear cat].my_any?(/d/) != %w[ant bear cat].any?(/d/)  ```
```  p [nil, true, 99].my_any?(Integer) != [nil, true, 99].any?(Integer) ```
```  p [nil, true, 99].my_any? != p [nil, true, 99].any?  ```